python/src/scipp/compat/mantid.py

```diff
@@ -359,32 +359,49 @@ def load(filename="",
 
     try:
         import mantid.simpleapi as mantid
-        from mantid.api import EventType
-        from mantid import AnalysisDataService
+        from mantid.api import Workspace
     except ImportError:
         raise ImportError(
             "Mantid Python API was not found, please install Mantid framework "
             "as detailed in the installation instructions (https://scipp."
             "readthedocs.io/en/latest/getting-started/installation.html)")
 
-    ws = mantid.Load(filename, **kwargs)
+    loaded = mantid.Load(filename, StoreInADS=False, **kwargs)
+
+    # Determine what Load has provided us
+    if isinstance(loaded, Workspace):
+        # A single workspace
+        data_ws = loaded
+        monitor_ws = None
+    else:
+        # Seperate data and monitor workspaces
+        data_ws = loaded.OutputWorkspace
+        monitor_ws = loaded.MonitorWorkspace
+
     if instrument_filename is not None:
-        mantid.LoadInstrument(ws,
+        mantid.LoadInstrument(data_ws,
                               FileName=instrument_filename,
                               RewriteSpectraMap=True)
 
     dataset = None
-    if ws.id() == 'Workspace2D':
-        dataset = convert_Workspace2D_to_dataset(ws)
-    elif ws.id() == 'EventWorkspace':
-        dataset = convert_EventWorkspace_to_dataset(ws, load_pulse_times,
-                                                    EventType)
-    elif ws.id() == 'TableWorkspace':
-        dataset = convert_TableWorkspace_to_dataset(ws, error_connection)
-
-    AnalysisDataService.Instance().remove(ws.name())
+    if data_ws.id() == 'Workspace2D':
+        has_monitors = False
+        for spec in data_ws.spectrumInfo():
+            has_monitors |= spec.isMonitor
```

Comment:
You could actually just terminate on the first occurrence of a monitor

Reply:
You can probably do this with a generator expression along the lines of 
`hasMonitors  =  next((item for item in specInfo if item.isMonitor), False)` (untested code)